src/writer.c

```diff
@@ -704,20 +704,37 @@ write_uri_node(SerdWriter* const writer,
   if (supports_abbrev(writer)) {
     if (field == FIELD_PREDICATE &&
         !strcmp((const char*)node->buf, NS_RDF "type")) {
-      return esink("a", 1, writer);
+	  if (writer->syntax == SERD_HEXTUPLES) {
+		  return esink("\"a\"", 3, writer);
+	  }
+	  else {
+		  return esink("a", 1, writer);
+	  }
     }
 
     if (!strcmp((const char*)node->buf, NS_RDF "nil")) {
-      return esink("()", 2, writer);
+	  if (writer->syntax == SERD_HEXTUPLES) {
```

Comment:
This is starting to push the limits of what makes sense in a single writer.  It's probably still (barely) fine, but probably time to split this thing up by syntax.